test/FileWriterTest.php

```diff
@@ -157,6 +160,24 @@ public function testUnwritableDirThrowsException() : void
         FileWriter::writeFile($dir . '/test', 'foo');
     }
 
+    public function testUnwritableDirThrowsExceptionWhenUsingCustomErrorHandler() : void
+    {
+        /** @psalm-suppress InvalidArgument */
+        set_error_handler(static function () : void {
+            throw new ErrorException();
+        });
+
+        $dir = sys_get_temp_dir() . '/unwritable';
+        touch($dir);
+
+        $this->expectException(RuntimeException::class);
```

Comment:
Don't we expect a `WriteContentException` here?

Reply:
No, because we got the exception from:

```php
if (dirname($tmp) !== realpath($dir)) {
    unlink($tmp);
    throw Exception\RuntimeException::unableToCreateTemporaryFile($dir);
}
```

as `/tmp/unwritable` is non writable and script fallback to `/tmp` so these paths are different.

Reply:
Ahh, never mind, I did not see that `RuntimeException` is part of this package, imported above :+1: